nwbinspector/checks/time_series.py

```diff
@@ -0,0 +1,85 @@
+"""Authors: Cody Baker, Ben Dichter, and Ryan Ly."""
+import numpy as np
+
+import pynwb
+
+# from ..tools import all_of_type
+from ..register_checks import register_check, Importance, Severity, InspectorMessage
+from ..utils import check_regular_series
+
+
+@register_check(importance=Importance.BEST_PRACTICE_VIOLATION, neurodata_type=pynwb.TimeSeries)
+def check_regular_timestamps(time_series: pynwb.TimeSeries, time_tol_decimals=9, gb_severity_threshold=1.0):
```

Comment:
```suggestion
def check_regular_timestamps(time_series: pynwb.TimeSeries, time_tol_decimals: int = 9, gb_severity_threshold: float = 1.0):
```
Just adding type hints for the additional parameters. Just a curious question, how are additional parameters on checks handled? Are additional parameters from checks exposed to the user so that they can be set via the command line?

Reply:
Sure, I'll go in and triple check all annotations where possible.

As for how those parameters get set by the user, right now they are not nor have we discussed how they should be; I'm treating them purely as internal developer-set values, with `check_function`-specific defaults relative to the severity thresholds for that specific check. So the only situation someone would actually use it in as an optional argument is manually applying one of the checks to a specific NWBFile object (outside command line use).

From my perspective, I am also trying to make such options as general as possible so that the same reserved option name can apply in the same manner across multiple checks, i.e., `gb_severity_threshold` could be a common threshold for any data size issues that effect severity. This means we could theoretically allow it to be specified at command line as a global option to apply to every check that takes that extra argument, but I believe we should think long and hard about that approach as over time it would undoubtedly lead to a large-scale accumulation of additional arguments, most of which a random user would likely not want to take the time to learn about or concern themselves with.